form/src/main/java/com/ritense/form/domain/FormIoFormDefinition.java

```diff
@@ -408,9 +408,14 @@ private static List<ArrayNode> getComponents(JsonNode formDefinition) {
     }
 
     private Object extractNodeValue(JsonNode node) {
-        if (node.isValueNode()) {
+        var nodeType = node.getNodeType();
+        if (nodeType == JsonNodeType.STRING) {
             return node.textValue();
-        } else if (node.isArray()) {
+        } else if (nodeType == JsonNodeType.NUMBER) {
+            return node.numberValue();
+        } else if (nodeType == JsonNodeType.BOOLEAN) {
+            return node.booleanValue();
+        } else if (nodeType == JsonNodeType.ARRAY) {
             List<String> values = new ArrayList<>();
             node.forEach(childNode -> values.add(childNode.textValue()));
```

Comment:
shouldn't this be some recursive method call, like
`node.forEach(childNode -> values.add(extractNodeValue(childNode)));`

?

Reply:
> shouldn't this be some recursive method call, like
This is mainly a bugfix and not a functional improvement. If you think this should be made recursive instead of letting us just update the whole array at once, then that can be discussed separately.

Reply:
Just looking at the code, the bug still exists when trying to access an array node with non-text values. So the bug would not be fully fixed as it is.

Reply:
Then this is a bigger issue in general. What if the array contains non-value nodes (objects)?
Recursion will add a lot of complication here and we only really should need to overwrite the whole array in the process variable.

I'm welcome to suggestions here, because we can not assume the type of every element in the array.

Reply:
>  It aims to resolve the issues the tribe is currently experiencing with one of their implementations 

It aims to solve an issue with pv. that is used in two implementations. This PR aims to solve valueNode parsing issues in this method, not containerNodes. containerNodes should've never been parsed in here to begin with since they need to be approached differently.

> I think we either need to decline the PR and figure out what exactly we want to do here, or look at the PR for what it is 

Both of these are fine with me as long as this issue gets resources assigned to it and not just put on the backlog.

Reply:
This PR does solve an issue, so thats good. Lets figure out what to do with the array some other time.